File: ext/internal/dto/message.go
package dto

import "sconcur/internal/types"

type Message struct {
	FlowKey string       `json:"fk" msgpack:"fk"`
	Method  types.Method `json:"md" msgpack:"md"`
	TaskKey string       `json:"tk" msgpack:"tk"`
	Payload []byte       `json:"pl" msgpack:"pl"`
	IsNext  bool         `json:"nx" msgpack:"nx"`
	// OwnerId is the opaque PHP-side coroutine id awaiting this task's result
	// (0 = nobody). Carried into the Result frame so the PHP scheduler routes
	// the result without its own task-to-fiber map.
	OwnerId int64 `json:"-" msgpack:"-"`
}

File: ext/src/dto/message.rs
//! Mirrors ext-go-legacy/internal/dto/message.go.

use crate::types::method::Method;

pub struct Message {
    pub flow_key: String,
    pub method: Method,
    pub task_key: String,
    pub payload: Vec<u8>,
    pub is_next: bool,
    /// The opaque PHP-side coroutine id awaiting this task's result (0 =
    /// nobody). Carried into the result frame so the PHP scheduler routes the
    /// result without its own task-to-fiber map.
    pub owner_id: i64,
}
